wp-includes/rest-api/endpoints/class-wp-rest-customize-sections-controller.php

```diff
@@ -0,0 +1,305 @@
+<?php
+/**
+ * REST API: WP_REST_Customize_Sections_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since ?.?.?
+ */
+
+/**
+ * Core class to access sections via the REST API.
+ *
+ * @since ?.?.?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Customize_Sections_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 4.7.0
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'customize/v1';
+		$this->rest_base = 'sections';
+	}
+
+	/**
+	 * Ensure customize manager.
+	 *
+	 * @return WP_Customize_Manager Manager.
+	 * @global WP_Customize_Manager $wp_customize
+	 */
+	public function ensure_customize_manager() {
+		global $wp_customize;
+		if ( empty( $wp_customize ) ) {
+			$wp_customize = new WP_Customize_Manager(); // WPCS: global override ok.
+		}
+		if ( ! did_action( 'customize_register' ) ) {
+
+			/** This action is documented in wp-includes/class-wp-customize-manager.php */
+			do_action( 'customize_register', $wp_customize );
+		}
+		$wp_customize->prepare_controls();
+		return $wp_customize;
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<section>[\w-|\[\]]+)', array(
+			'args' => array(
+				'section' => array(
+					'description' => __( 'Identifier for the section.' ),
+					'type'        => 'string',
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => array(
+					'context' => $this->get_context_param( array(
+						'default' => 'view',
+					) ),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Retrieves the section's schema, conforming to JSON Schema.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+
+		$schema = array(
```

Comment:
Since the `json` methods in `WP_Customize_Setion` subclass can add additional params, it would probably be good to explicitly add the `'additionalProperties' => true,` to the schema. That would mean that the additional props could be included even if the `WP_Customize_Section` didn't register the field's schema via `register_rest_field()`.